### ISSUE TITLE
Remove first bar causes a crash

### ISSUE THREAD

**Issue by zorba77** [Jojo-Schmitz/MuseScore#414]
If I:
- select the first bar
-  right click
- remove the bar
Musescore 3.7 crashes. No problems with MS 3.6.2 or 4.X
The crash occurs if the first bar is empty

**Comment by Jojo-Schmitz** [comment]
"remove the bar" is not an entry in the right-click menu, do you mean "Remove Selected Range" or "Delete"?
I can't reproduce the crash with either, can you provide a sample score (zipped, as otherwish you can't attach it here)?

**Comment by zorba77** [comment]
here the score

Il giorno dom 14 apr 2024 alle ore 12:01 Joachim Schmitz <
***@***.***> ha scritto:

**Comment by Jojo-Schmitz** [comment]
No, it isn't

**Comment by Jojo-Schmitz** [comment]
Also I'd rather have all my questions answered

**Comment by zorba77** [comment]
Sorry
To reproduce:
1. select the first bar
2. right click
3. "Remove Selected Range"
It crashes
OS: Ubuntu 22.04.4 LTS, Arch.: x86_64, MuseScore version (64-bit): 3.7.0.8678350200, revision: GitHub-Jojo-Schmitz-MuseScore-acbfac0

[Esseri_Umani_AB_OK ERROR.zip](https://github.com/Jojo-Schmitz/MuseScore/files/14982401/Esseri_Umani_AB_OK.ERROR.zip)

**Comment by Jojo-Schmitz** [comment]
A Debug build dies at https://github.com/Jojo-Schmitz/MuseScore/blob/c2535e4797b4caf857f5cc5411e1f68e74094f8c/libmscore/segment.cpp#L507

**Comment by Jojo-Schmitz** [comment]
Possible culprit (the only new thing in the stack trace): 1bbee5a

**Comment by Jojo-Schmitz** [comment]
And indeed reverting that fixes the crash...

**Comment by Jojo-Schmitz** [comment]
The only difference between that 1bbee5a and 09effa2 (the one I backporeted) is:
```cpp
                        TimeSig* lastDeletedForThisStaff = nullptr;
                        if (lastDeletedTimeSigs.find(staffIdx) != lastDeletedTimeSigs.end())
                              lastDeletedForThisStaff = lastDeletedTimeSigs.find(staffIdx)->second;
                        else
                              continue;
```
vs.
```cpp
                TimeSig* lastDeletedForThisStaff = mu::value(lastDeletedTimeSigs, staffIdx, nullptr);
                if (!lastDeletedForThisStaff) {
                    continue;
                }
```

**Comment by Jojo-Schmitz** [comment]
A possible fix is known: reverting 1bbee5a, but I'd rather not fix one crash by reintroducing another

**Comment by worldwideweary** [comment]
Think maybe just setting the score while adding the new element would fix the problem?

**Comment by Jojo-Schmitz** [comment]
Huh? I'm affraid you lost me here...

**Comment by worldwideweary** [comment]
I'm not sure, I was just thinkin' out loud. It crashes here:
```cpp
void Segment::add(Element* el)
      {
//      qDebug("%p segment %s add(%d, %d, %s)", this, subTypeName(), tick(), el->track(), el->name());

      el->setParent(this);

      int track = el->track();
      Q_ASSERT(track != -1);
******Q_ASSERT(el->score() == score());   <---------------------------
      Q_ASSERT(score()->nstaves() * VOICES == int(_elist.size()));
```

**Comment by Jojo-Schmitz** [comment]
Pretty much the same code in  Mu4

**Comment by worldwideweary** [comment]
Looks like someone threw in a Q_ASSERT for something that doesn't allow for bypassing the adding of a time signature but merely cracks the program, so we can test for it before that happens and just not add it in there

**Comment by Jojo-Schmitz** [comment]
Fixed via #637
